utils/containers/hash/hash_array.c

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) 2018 IOTA Stiftung
+ * https://github.com/iotaledger/entangled
+ *
+ * Refer to the LICENSE file for licensing information
+ */
+
+#include "utils/containers/hash/hash_array.h"
+
+static UT_icd ut_hash27_icd = {FLEX_TRIT_SIZE_27, NULL, NULL, NULL};
+static UT_icd ut_hash81_icd = {FLEX_TRIT_SIZE_81, NULL, NULL, NULL};
+static UT_icd ut_hash243_icd = {FLEX_TRIT_SIZE_243, NULL, NULL, NULL};
+static UT_icd ut_hash6561_icd = {FLEX_TRIT_SIZE_6561, NULL, NULL, NULL};
+static UT_icd ut_hash8019_icd = {FLEX_TRIT_SIZE_8019, NULL, NULL, NULL};
+
+hash27_array_p hash27_array_new() {
+  hash27_array_p array = NULL;
+  utarray_new(array, &ut_hash27_icd);
+  return array;
+}
+
+hash81_array_p hash81_array_new() {
+  hash81_array_p array = NULL;
+  utarray_new(array, &ut_hash81_icd);
+  return array;
+}
+
+hash243_array_p hash243_array_new() {
+  hash243_array_p array = NULL;
+  utarray_new(array, &ut_hash243_icd);
+  return array;
+}
+
+hash6561_array_p hash6561_array_new() {
+  hash6561_array_p array = NULL;
+  utarray_new(array, &ut_hash6561_icd);
+  return array;
+}
+
+hash8019_array_p hash8019_array_new() {
+  hash8019_array_p array = NULL;
+  utarray_new(array, &ut_hash8019_icd);
+  return array;
+}
+
+void hash_array_push(hash_array_p array, flex_trit_t const* const trits) {
```

Comment:
Given our discussion just now on Slack, could you add a `hash_array_reserve` method here?

Reply:
done